witchcraft-server/src/configs.rs

```diff
@@ -46,15 +51,60 @@ where
 {
     let key = load_key()?;
     let bytes = load_file(RUNTIME_YML)?;
-    let value = parse(&bytes, key.as_ref())?;
+    let (value, files) = parse(&bytes, key.as_ref());
+    let value = value?;
 
     let (refreshable, handle) = Refreshable::new(value);
 
-    runtime.spawn(runtime_reload(bytes, key, handle, config_ok.clone()));
+    runtime.spawn(runtime_reload(files, key, handle, config_ok.clone()));
 
     Ok(refreshable)
 }
 
+struct ConfigFiles {
+    root_hash: Output<Sha256>,
+    ok_files: HashMap<PathBuf, Output<Sha256>>,
+    err_files: HashSet<PathBuf>,
+}
+
+impl ConfigFiles {
+    fn up_to_date(&self, root_bytes: &[u8]) -> bool {
+        if self.root_hash != Sha256::digest(root_bytes) {
+            return false;
+        }
+
+        for (path, hash) in &self.ok_files {
+            match fs::read(path) {
+                Ok(bytes) => {
+                    if *hash != Sha256::digest(&bytes) {
+                        return false;
+                    }
+                }
+                Err(_) => return false,
+            }
+        }
+
+        for path in &self.err_files {
+            if fs::read(path).is_ok() {
```

Comment:
We could alternatively say that we're never up to date when we have failed file references.

Reply:
I think it's probably preferable to avoid the downstream churn if we just have some file that we know is not valid, so this seems right